### PR TITLE
[Example] Variational Graph Auto-Encoders

### PR DESCRIPTION
**Pull request by JuliaSun623**
## Description
This PR is the implementation of the paper: Variational Graph Auto-Encoders.

I currently included two preprocess methods of dataset. One is using DGL's built-in dataset and another is using the method used in original paper. Results shows that using DGL's dataset get the similar result reported in the paper, while another method has worse performance. 

The method using DGL's built-in dataset can be trained on GPU.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
Can you also update the indexing page [here](https://github.com/dmlc/dgl/tree/master/examples)?

**Comment by JuliaSun623** [comment]
Already update the indexing page

File: examples/pytorch/vgae/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Variational Graph Auto-Encoders
+
+- Paper link：https://arxiv.org/abs/1611.07308
+- Author's code repo：https://github.com/tkipf/gae
+
+## Requirements
+
+- Pytorch
+- Python 3.x
+- DGL
```
[the remark this answers is not in the document]
Comment (JuliaSun623) [reply examples/pytorch/vgae/README.md]: Already added.

File: examples/pytorch/vgae/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+| Dataset  | AUC            | AP            |
+| -------- | -------------- | ------------- |
+| Cora     | 91.8$\pm$ 0.01 | 92.5$\pm$0.01 |
+| Citeseer | 87.6$\pm$0.01  | 89.6$\pm$0.01 |
+| Pubmed   | 89.6$\pm$0.01  | 91.7$\pm$0.01 |
+
+### Dataset from website
+
+| Dataset  | AUC            | AP            |
```
[the remark this answers is not in the document]
Comment (JuliaSun623) [reply examples/pytorch/vgae/README.md:48]: Already added.

**Comment by sneakerkg** [comment]
Why we close this PR?

**Comment by JuliaSun623** [comment]
This PR is transform to a new one for I made changes on "master" branch by mistake. New PR link: https://github.com/dmlc/dgl/pull/2587. These two are the same.

**Comment by JuliaSun623** [comment]
> Why we close this PR?

This PR is transformed to a new one, for I made changes on "master" branch by mistake. New PR link: https://github.com/dmlc/dgl/pull/2587. These two are the same.

**Pull request closed without merging**